state/machine.go

```diff
@@ -2020,6 +2026,58 @@ func (m *Machine) UpdateMachineSeries(series string, force bool) error {
 	return errors.Annotatef(err, "cannot update series for %q to %s", m, series)
 }
 
+// CreateUpgradeSeriesPrepareLock create a prepare lock for series upgrade. If
+// this item exists in the database for a given machine it indicates that a
+// machine's operating system is being upgraded for one series to another (e.g. xenial to bionic).
+func (m *Machine) CreateUpgradeSeriesPrepareLock() error {
+	locked, err := m.IsLocked()
```

Comment:
This should be checked inside of buildTxn.... after attempt, then if locked {
return nil, jujutxn.ErrNoOperations
}